geode-redis/src/main/java/org/apache/geode/redis/internal/data/RedisString.java

```diff
@@ -610,18 +613,26 @@ public int setbit(
    * and call setBytes. So toData will see either the old or new value but
    * not a mix of both.
    */
+
   @Override
-  public void toData(DataOutput out) throws IOException {
-    super.toData(out);
+  public synchronized void toData(DataOutput out, SerializationContext context) throws IOException {
```

Comment:
toData has a lengthy comment explaining why it does not need to be synchronized. It would be good to update that comment to explain why it needs to be synchronized

Reply:
updated